azurerm/common_hdinsight.go

```diff
@@ -57,6 +60,42 @@ func hdinsightClusterUpdate(clusterKind string, readFunc schema.ReadFunc) schema
 			}
 		}
 
+		// The API can add an edge node but can't remove them without force newing the resource. We'll check for adding here
+		// and can come back to removing if that functionality gets added. https://feedback.azure.com/forums/217335-hdinsight/suggestions/5663773-start-stop-cluster-hdinsight?page=3&per_page=20
+		if clusterKind == "Hadoop" {
+			if d.HasChange("roles.0.edge_node") {
+				log.Printf("[DEBUG] Detected change in edge nodes")
+				edgeNodeRaw := d.Get("roles.0.edge_node").([]interface{})
+				edgeNodeConfig := edgeNodeRaw[0].(map[string]interface{})
+				applicationsClient := meta.(*ArmClient).HDInsight.ApplicationsClient
+
+				// Note: API currently doesn't support updating number of edge nodes
+				// if anything in the edge nodes changes, delete edge nodes then recreate them
+				err := deleteHDInsightEdgeNodes(ctx, applicationsClient, resourceGroup, name)
```

Comment:
We should confirm this actually works by updating an edge node in a test. [CustomizeDiff](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4550/files#diff-29907bf0f1246f030e1ee8af21a80056R64) will currently tear down and spin up a new resource so that will have to be looked at as well before this can be properly tested.

Reply:
I've determined that CustomizeDiff is not necessary because we don't need to tear down the whole cluster for edge node changes. This logic takes care of it correctly.